src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -41,21 +43,125 @@ export const Breadcrumbs = ({
   ariaLabel = 'breadcrumbs links',
   ...other
 }: Props) => {
+  const [shouldTruncate, setShouldTruncate] = React.useState(false);
+
+  const ref = React.useRef<HTMLUListElement>(null);
+
+  /**
+   * Checks if breadcrumbs would overflow
+   */
+  const updateShouldTruncate = () => {
+    setShouldTruncate(false);
+    if (
+      ref &&
+      ref.current &&
+      ref.current.clientWidth < ref.current.scrollWidth
+    ) {
```

Comment:
Can we do the chainyquestionythingy?
```suggestion
    if (ref?.current?.clientWidth < ref.current.scrollWidth) {
```

Reply:
Doesn't work :( 
![image](https://user-images.githubusercontent.com/86632227/173932697-1b524f90-faed-4861-a49b-b5ade699273f.png)

Reply:
Weird 🤔

Reply:
I'm fine with leaving it as-is, but in case you're curious, this might explain why it's not working: https://stackoverflow.com/questions/66017941/object-is-possibly-undefined-error-on-optional-chaining-typescript

Reply:
Yea it's because the initial value is `null`. I changed it to empty object so the optional chaining works now, thanks!

Reply:
Ah TIL about nullish colescing and optional chaining and TS lol. Changed to:
```tsx
  if ((ref?.current?.clientWidth ?? 0) < (ref?.current?.scrollWidth ?? 0)) {
```

Reply:
I think what you had originally is easier to read though, even if it takes up multiple lines with the `&&` operators.

Reply:
Same, will change back